os/drivers/video/video.c

```diff
@@ -315,8 +316,10 @@ static void change_video_state(FAR video_upperhalf_t *priv, enum video_state_e n
 	if ((current_state != VIDEO_STATE_DMA) && (next_state == VIDEO_STATE_DMA)) {
 		dma_container = video_framebuff_get_dma_container(&priv->video_inf.bufinf);
 		if (dma_container) {
-			video_devops->set_buftype(priv->dev, V4L2_BUF_TYPE_VIDEO_CAPTURE);
-			video_devops->set_buf(priv->dev, dma_container->buf.m.userptr, dma_container->buf.length);
+			ret = video_devops->set_buf(priv->dev, V4L2_BUF_TYPE_VIDEO_CAPTURE, dma_container->buf.m.userptr, dma_container->buf.length);
+			if (ret < OK) {
+				return ret;
+			}
 		} else {
 			updated_next_state = VIDEO_STATE_STREAMON;
 		}
```

Comment:
how about changing like below?
line 319
ret = video_devops->...
 [remove 320~322 line]

line 332
if (ret == OK) {
  priv->video_inf.state = updated_next_state;
}
return ret;

Reply:
Corrected in set_buftype & set_buf unify patch.